bluemira/builders/EUDEMO/maintenance.py

```diff
@@ -0,0 +1,268 @@
+        x2 = x - np.sin(np.deg2rad(gamma))
+        y2 = y
+        z2 = z + np.cos(np.deg2rad(gamma))
+        angled_cut_plane = BluemiraPlane.from_3_points(
+            intersection, [x2, y2, z2], [x, y + 1, z]
+        )
+        # Get the last intersection with the angled cut plane and the outer
+        intersections = slice_shape(bb.boundary[0], angled_cut_plane)
+        intersections = intersections[intersections[:, -1] > z + EPS]
+        intersection = sorted(intersections, key=lambda x: x[-1])[0]
+        return intersection
+
+    def optimise(self, x0=None):
+        """
+        Solve the optimisation problem.
+        """
+        if x0 is None:
+            R_0 = self.params.R_0.value
+            x0 = np.array([R_0, R_0, R_0, 0])
+        return self.opt.optimise(x0)
+
+
+def segment_blanket_xz(breeding_blanket_xz, r_inner_cut, cut_angle, cut_thickness):
+    """
+    Segment the breeding blanket poloidal cross-section into inboard and outboard
+    segment silhouettes.
+
+    Parameters
+    ----------
+    breeding_blanket_xz: BluemiraFace
+        Breeding blanket poloidal cross-section (unsegmented)
+    r_inner_cut: float
+        Cut radius on the plasma-facing surface
+    cut_angle: float
+        Cut plane angle (off from vertical) [degrees]
+    cut_thickness: float
+        Thickness of the cut zone
+
+    Returns
+    -------
+    ib_silhouette: BluemiraFace
+        Inboard blanket segment silhouette
+    ob_silhouette: BluemiraFace
+        Outboard blanket segment silhouette
+    """
+    # Make cutting geometry
+    p0 = _get_inner_cut_point(breeding_blanket_xz, r_inner_cut)
+    p1 = [p0[0], 0, p0[2] + VERY_BIG]
+    p2 = [p0[0] - cut_thickness, 0, p1[2]]
+    p3 = [p2[0], 0, p0[2] - np.sqrt(2) * cut_thickness]
+    cut_zone = BluemiraFace(make_polygon([p0, p1, p2, p3], closed=True))
+    if cut_angle != 0.0:
+        cut_zone.rotate(base=p0, direction=(0, -1, 0), degree=cut_angle)
+
+    # Do cut
+    cut_result = boolean_cut(breeding_blanket_xz, cut_zone)
+    if len(cut_result) < 2:
+        raise BuilderError(
+            f"BB poloidal segmentation only returning {len(cut_result)} faces."
+        )
+    if len(cut_result) > 2:
+        bluemira_warn(
+            f"The BB poloidal segmentation operation returned more than 2 faces ({len(cut_result)}); only taking the first two..."
+        )
+    ib_silhouette, ob_silhouette = sorted(cut_result, key=lambda x: x.center_of_mass[0])[
+        :2
+    ]
+    return ib_silhouette, ob_silhouette
+
+
+def build_upper_port_zone(r_up_inner, r_up_outer, z_max=10, z_min=0):
+    """
+    Make the void geometry for the upper port in the poloidal plane.
+    """
+    x = [r_up_inner, r_up_outer, r_up_outer, r_up_inner]
+    z = [z_min, z_min, z_max, z_max]
+    return BluemiraFace(make_polygon({"x": x, "y": 0, "z": z}, closed=True))
+
+
+if __name__ == "__main__":
+    # TODO: Remove this once threaded into the reactor build via some kind of builder
```

Comment:
Do we have an issue that resolves this TODO? If there's an existing issue, is it explicit about removing this?